### PR TITLE
Fix not returning mp3 file with yt-dlp

### PR DESCRIPTION
**Pull request by MichaelBelgium**
Issue noticed in https://github.com/norkunas/youtube-dl-php/issues/162#issuecomment-1008836480
Tested the regex here: https://regex101.com/r/sqwPFw/1

Tested and returned a var dump of `getFile()` and returns
```
object(SplFileInfo)#9 (2) {
  ["pathName":"SplFileInfo":private]=>
  string(24) "download/_VeNOH9i06I.mp3"
  ["fileName":"SplFileInfo":private]=>
  string(15) "_VeNOH9i06I.mp3"
}
```

### CODE REVIEW

**Comment by norkunas** [comment]
Would you be able to add a test covering this?

**Comment by norkunas** [comment]
Yes, you're on the right path

**Comment by norkunas** [comment]
Now you should also add cases to YoutubeDlTest.php using that outputs :)

File: tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt [hunk 1]
```diff
@@ -0,0 +1,64 @@
+[youtube] oDAw7vW7H0c: Downloading webpage
+[info] Writing video description metadata as JSON to: ./oDAw7vW7H0c.info.json
```
Comment (norkunas) [inline tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt:2]: And as in the tests virtual filesystem is used, then this path should be:

```suggestion
[info] Writing video description metadata as JSON to: vfs://yt-dl/oDAw7vW7H0c.info.json
```

File: tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+[download] Destination: ./oDAw7vW7H0c.webm
```
Comment (norkunas) [inline tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt:3]:
```suggestion
[download] Destination: vfs://yt-dl/oDAw7vW7H0c.webm
```

File: tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+[download]  87.4% of 2.69MiB at 52.51KiB/s ETA 00:06
+[download]  89.5% of 2.69MiB at 52.36KiB/s ETA 00:05
+[download]  91.2% of 2.69MiB at 52.33KiB/s ETA 00:04
+[download]  93.0% of 2.69MiB at 52.34KiB/s ETA 00:03
+[download]  94.9% of 2.69MiB at 52.45KiB/s ETA 00:02
+[download]  97.0% of 2.69MiB at 52.34KiB/s ETA 00:01
+[download]  98.8% of 2.69MiB at 52.34KiB/s ETA 00:00
+[download] 100.0% of 2.69MiB at 52.56KiB/s ETA 00:00
+[download] 100% of 2.69MiB in 00:52
+[ffmpeg] Destination: ./oDAw7vW7H0c.mp3
```
Comment (norkunas) [inline tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt:63]:
```suggestion
[ffmpeg] Destination: vfs://yt-dl/oDAw7vW7H0c.mp3
```

File: tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Deleting original file ./oDAw7vW7H0c.webm (pass -k to keep)
```
Comment (norkunas) [inline tests/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt:64]:
```suggestion
Deleting original file vfs://yt-dl/oDAw7vW7H0c.webm (pass -k to keep)
```

File: tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+[youtube] oDAw7vW7H0c: Downloading webpage
+[youtube] oDAw7vW7H0c: Downloading android player API JSON
+[info] oDAw7vW7H0c: Downloading 1 format(s): 251
+[info] Writing video metadata as JSON to: ./oDAw7vW7H0c.info.json
```
Comment (norkunas) [inline tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt:4]:
```suggestion
[info] Writing video metadata as JSON to: vfs://yt-dl/oDAw7vW7H0c.info.json
```

File: tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+[download] Destination: ./oDAw7vW7H0c.webm
```
Comment (norkunas) [inline tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt:5]:
```suggestion
[download] Destination: vfs://yt-dl/oDAw7vW7H0c.webm
```

File: tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+[download]   1.1% of 2.69MiB at 806.37KiB/s ETA 00:03
+[download]   2.3% of 2.69MiB at 1016.86KiB/s ETA 00:02
+[download]   4.6% of 2.69MiB at  1.32MiB/s ETA 00:01  
+[download]   9.2% of 2.69MiB at  2.01MiB/s ETA 00:01
+[download]  18.5% of 2.69MiB at  2.68MiB/s ETA 00:00
+[download]  37.1% of 2.69MiB at  2.33MiB/s ETA 00:00
+[download]  74.2% of 2.69MiB at  3.90MiB/s ETA 00:00
+[download] 100% of 2.69MiB at  4.05MiB/s ETA 00:00  
+[download] 100% of 2.69MiB in 00:00               
+[ExtractAudio] Destination: ./oDAw7vW7H0c.mp3
```
Comment (norkunas) [inline tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt:20]:
```suggestion
[ExtractAudio] Destination: vfs://yt-dl/oDAw7vW7H0c.mp3
```

File: tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+Deleting original file ./oDAw7vW7H0c.webm (pass -k to keep)
```
Comment (norkunas) [inline tests/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt:21]:
```suggestion
Deleting original file vfs://yt-dl/oDAw7vW7H0c.webm (pass -k to keep)
```

**Review by norkunas — CHANGES_REQUESTED** [review]

**Comment by MichaelBelgium** [comment]
> Now you should also add cases to YoutubeDlTest.php using that outputs :)

Well it never can be _exactly_ the same right? Like the download speeds and stuff?

I add a function `testDownloadMp3` or something like that? Like [here](https://github.com/norkunas/youtube-dl-php/blob/master/tests/YoutubeDlTest.php#L178-L195)
Or how to say it needs to assert that output with `download()` ... etc

You can tell i don't write a lot of tests 😂

**Comment by norkunas** [comment]
Download speeds doesn't matter :) it's just enough to ensure that it can properly parse the output and create a video collection from the downloaded metadata file

**Comment by MichaelBelgium** [comment]
I came up with this test but its probably not the right approach.

When executing `vendor\bin\simple-phpunit` it says it's a risky test and doesn't assert anything?


```
public function provideMp3VideoFile(): iterable
    {
        $expectedFile = new SplFileInfo('vfs://yt-dl/oDAw7vW7H0c.mp3');

        yield 'youtube-dl: phonebloks' => [
            'url' => 'https://www.youtube.com/watch?v=oDAw7vW7H0c',
            'outputFile' => __DIR__.'/Fixtures/youtube-dl/youtube/phonebloks-mp3.txt',
            'expectedEntity' => new Video(['file' => $expectedFile]),
        ];

        yield 'yt-dlp: phonebloks' => [
            'url' => 'https://www.youtube.com/watch?v=oDAw7vW7H0c',
            'outputFile' => __DIR__.'/Fixtures/yt-dlp/youtube/phonebloks-mp3.txt',
            'expectedEntity' => new Video(['file' => $expectedFile]),
        ];
    }
```

**Comment by norkunas** [comment]
Tests looks fine :) just add the metadata files to fix them

**Comment by MichaelBelgium** [comment]
Isn't the metafile already there? [This](https://github.com/norkunas/youtube-dl-php/blob/master/tests/Fixtures/youtube-dl/youtube/Phonebloks-oDAw7vW7H0c.info.json) one, Or am i missing something

I don't get the phpunit error

**Comment by norkunas** [comment]
It's there, but it is used as for video download, not for mp3. This library builds the VideoCollection/Video from the metadata, but before doing that it also appends `_filename` to the end file, so to mimick that we need the new one metadata file for mp3 which you'd get to your device if you'd run with `cleanupMetadata(false)` option.

**Comment by MichaelBelgium** [comment]
So i just replace the old one with the mp3 metadata one?

**Comment by norkunas** [comment]
I've fixed the tests, but cannot push to your PR. Did you enable this?:
image

**Comment by MichaelBelgium** [comment]
Now i did :)

**Review by norkunas — APPROVED** [review]

**Comment by norkunas** [comment]
Thank you :)

**Pull request merged**